go/merge/three_way_keyval_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package merge
+
+import (
+	"testing"
+
+	"github.com/attic-labs/noms/go/types"
+	"github.com/attic-labs/testify/suite"
+)
+
+func TestThreeWayMapMerge(t *testing.T) {
+	suite.Run(t, &ThreeWayMapMergeSuite{})
+}
+
+func TestThreeWayStructMerge(t *testing.T) {
+	suite.Run(t, &ThreeWayStructMergeSuite{})
+}
+
+type kvs []interface{}
+
+func (kv kvs) items() []interface{} {
+	return kv
+}
+
+func (kv kvs) remove(k interface{}) kvs {
+	out := make(kvs, 0, len(kv))
+	for i := 0; i < len(kv); i++ {
```

Comment:
wait shouldn't this be i += 2? otherwise you'll remove map values that equal `k` and boom.